internal/services/conversation_optimizer.go

```diff
@@ -7,45 +7,51 @@ import (
 	"time"
 
 	"github.com/inference-gateway/cli/config"
+	"github.com/inference-gateway/cli/internal/models"
 	"github.com/inference-gateway/sdk"
 )
 
 // ConversationOptimizer provides methods to optimize conversation history for token efficiency
 type ConversationOptimizer struct {
-	enabled     bool
-	model       string
-	minMessages int
-	bufferSize  int
-	client      sdk.Client
-	config      *config.Config
+	enabled    bool
+	autoAt     int // percentage of context window (20-100)
+	bufferSize int
+	client     sdk.Client
+	config     *config.Config
+	tokenizer  *TokenizerService
 }
 
 // OptimizerConfig represents configuration for the conversation optimizer
 type OptimizerConfig struct {
-	Enabled     bool
-	Model       string
-	MinMessages int
-	BufferSize  int
-	Client      sdk.Client
-	Config      *config.Config
+	Enabled    bool
+	AutoAt     int // percentage of context window (20-100)
+	BufferSize int
+	Client     sdk.Client
+	Config     *config.Config
+	Tokenizer  *TokenizerService
 }
 
 // NewConversationOptimizer creates a new conversation optimizer with configuration
 func NewConversationOptimizer(config OptimizerConfig) *ConversationOptimizer {
-	if config.MinMessages <= 0 {
-		config.MinMessages = 10
+	if config.AutoAt < 20 || config.AutoAt > 100 {
+		config.AutoAt = 80
```

Comment:
In this if block there should be a warning log indicating that AutoAt can only be between the range and that it is defaulting to 80 because the value that was set is incorrect.